qt/src/hocr/HOCRPdfExporter.cc

```diff
@@ -283,6 +290,23 @@ HOCRPdfExporter::HOCRPdfExporter(const HOCRDocument* hocrdocument, const HOCRPag
 	ui.comboBoxImageCompression->addItem(_("Jpeg (lossy)"), PDFSettings::CompressJpeg);
 	ui.comboBoxImageCompression->setCurrentIndex(-1);
 
+	ui.comboBoxPaperSizeMetric->addItem(_("cm"), static_cast<int>(PaperSizeMetrics::cm));
+	ui.comboBoxPaperSizeMetric->addItem(_("in"), static_cast<int>(PaperSizeMetrics::inch));
+
+	ui.comboBoxPaperSize->addItem(_("Same as source"), "Same as source");
+	ui.comboBoxPaperSize->addItem(_("Custom"), "Custom");
+	for(const auto& size : PaperSizeProcessor::paperSizes) {
+		ui.comboBoxPaperSize->addItem(size.first.c_str(), size.first.c_str());
+	}
+
+	ui.lineEditPaperSizeHeight->setValidator(new QDoubleValidator(0.0, 10000000.0, 2));
+	ui.lineEditPaperSizeWidth->setValidator(new QDoubleValidator(0.0, 10000000.0, 2));
+
+	if(ui.comboBoxPaperSize->itemData(ui.comboBoxPaperSize->currentIndex()).toString() == "Same as source") {
```

Comment:
This isn't 100% correct, the fields should only be enabled in custom mode, and disabled in same as source and fixed-size-template modes. Also, I'd store it as a setting: add a setCurrentIndex(-1) here, to ensure no entry is selected, then add a ComboSetting below, which will then take care of restoring the previous choice, and the change of index triggered by the setting whill then call paperSizeChanged which will enable/disable the line edits as necessary.

Reply:
Ok. I also prefer add to settings PaperSizeMetric combobox.

Reply:
Fixed.